docs/shrc.md

```diff
@@ -66,43 +140,138 @@ The [-Xshareclasses](xshareclasses.md) option is highly configurable, allowing y
 
     In addition, if you have VMs from different Java installations, of the same Java release and installed by the same user, each VM checks whether the existing default shared cache in the cache directory is from the same Java installation as the VM. If not, the VM deletes that shared cache, then creates a new one. Specifying a different cache directory for each Java installation avoids this situation.
 
-
-
 - Ensure that the cache directory permissions are set appropriately ([`-Xshareclasses:cacheDirPerm`](xshareclasses.md#cachedirperm)).
 
-    It is good practice to explicitly set permissions for the cache directory when the defaults are not appropriate.
+    It is good practice to explicitly set permissions for the cache directory when the defaults are not appropriate. Access is controlled by operating system permissions and Java security permissions; read/write access is the default and gives all users equal rights to update the cache. On Unix systems, you can use the [`-Xshareclasses:groupAccess`](xshareclasses.md#groupaccess) suboption to allow read/write permissions for groups as well as users.
+
+    On z/OS, a cache can be accessed only by a VM that is running in the same storage key as the VM that created the cache. If the keys do not match, permission to access the cache is denied.
 
 - Set the [`-Xshareclasses:nonfatal`](xshareclasses.md#nonfatal) option.
 
-    This option means that your application can start even if there is a problem opening or creating the shared cache, in which case, the VM might be able to start *without* class data sharing.
+    Setting this option allows your application to start even if there is a problem opening or creating the shared cache. In this situation, the VM might be able to start *without* class data sharing.
```

Comment:
We should note here or in the `nonfatal` doc that this option doesn't address all cases, just certain scenarios such as not having read access to an existing cache, or detectable corruption. i.e. it doesn't help with bugs or all cache corruption. Cache corruption (generally as a result of a bug in the OS, VM, user code) isn't necessarily detected when opening the cache, the cache will be used and the application may crash.

Reply:
I don't think this should say "might be able", but that the VM will continue to start without the shared cache.

Reply:
Changed to:

*In most cases, setting this option allows your application to start even if there is a problem opening or creating the shared cache. The VM will continue to start *without* class data sharing.*

I added the extra info you provided into the `xshareclasses.md` topic as a note to the `nonfatal` suboption.